explorer/frontend/default.nix

```diff
@@ -1,26 +1,121 @@
-if builtins.compareVersions "1.11.7" builtins.nixVersion == 1 then
-  abort ''
-    This project requires Nix >= 1.11.7, please upgrade:
+let
+  localLib = import ../../lib.nix;
+in
+{ system ? builtins.currentSystem
+, config ? {}
+, gitrev ? localLib.commitIdFromGitRepo ../../.git
+, pkgs ? (import (localLib.fetchNixPkgs) { inherit system config; })
+, cardano-sl-explorer
+}:
 
-       curl https://nixos.org/nix/install | sh
-  ''
-else
+with pkgs.lib;
 
-with (import (fetchTarball https://github.com/NixOS/nixpkgs/archive/cb90e6a0361554d01b7a576af6c6fae4c28d7513.tar.gz) {});
+let
+  cleanSourceFilter = with pkgs.stdenv;
+    name: type: let baseName = baseNameOf (toString name); in ! (
+      # Filter out .git repo
+      (type == "directory" && baseName == ".git") ||
+      # Filter out editor backup / swap files.
+      lib.hasSuffix "~" baseName ||
+      builtins.match "^\\.sw[a-z]$" baseName != null ||
+      builtins.match "^\\..*\\.sw[a-z]$" baseName != null ||
 
-# https://github.com/paf31/purescript-derive-lenses/issues/12
-# cabal2nix https://github.com/paf31/purescript-derive-lenses.git > purescript-derive-lenses.nix
+      # Filter out locally generated/downloaded things.
+      baseName == "bower_components" ||
+      (type == "directory" && (baseName == "node_modules" || baseName == "dist")) ||
 
-let
-  hspkgs = pkgs.haskell.packages.ghc802.override {
+      # Filter out the files which I'm editing often.
+      lib.hasSuffix ".nix" baseName ||
+      # Filter out nix-build result symlinks
+      (type == "symlink" && lib.hasPrefix "result" baseName)
+    );
+
+  src = builtins.filterSource cleanSourceFilter ./.;
+
+  bowerComponents = pkgs.buildBowerComponents {
+    name = "cardano-sl-explorer-frontend-deps";
+    generated = ./bower-generated.nix;
+    inherit src;
+  };
+
+  nodePackages = import ./composition.nix {
+    inherit pkgs system;
+    inherit src;
+  };
+
+  # p-d-l does not build with our main version of nixpkgs.
+  # Needs to use something off 17.03 branch.
+  oldHaskellPackages = (import (fetchTarball https://github.com/NixOS/nixpkgs/archive/cb90e6a0361554d01b7a576af6c6fae4c28d7513.tar.gz) {}).pkgs.haskell.packages.ghc802.override {
```

Comment:
Can we use `fetch-nixpkgs.nix` here from 17.09? We want to avoid having mulitple nixpkgs versions around.

Reply:
`purescript-derive-lenses` doesn't compile with the nixpkgs used everywhere else. So it needs to nixpkgs to stay as is. We could update `purescript-derive-lenses` to fix?

Reply:
I have dug into this again because having multiple nixpkgs isn't good.

 - Changed it to use the ghc802 packageset from 17.09 (same as everything else).
 - When calling `purescript-derive-lenses` package it needs a jailbreak (because of `optparse-generic` version)
 - Then there is a build error
   ```
   src/Main.hs:66:18: error:
       Not in scope: data constructor ‘P.PositionedDeclaration’
       Module ‘Language.PureScript’ does not export ‘PositionedDeclaration’.
   ```

To fix this would probably require either a fork of [purescript-derive-lenses](https://github.com/paf31/purescript-derive-lenses) (the new version is written in purescript now), or trying to override the purescript version to something older than 0.11.6 (maybe 0.10.*).